kani-compiler/src/args.rs

```diff
@@ -69,11 +82,12 @@ pub struct Arguments {
     /// Pass the kani version to the compiler to ensure cache coherence.
     check_version: Option<String>,
     #[clap(long)]
-    /// A legacy flag that is now ignored.
-    goto_c: bool,
-    /// Enable specific checks.
-    #[clap(long)]
     pub ub_check: Vec<ExtraChecks>,
+    #[clap(long = "backend")]
+    pub backend: Option<BackendOption>,
```

Comment:
Please document. Also, what do you think about:

```suggestion
    #[clap(long = "backend", default_value_t = BackendOption::CProver)]
    pub backend: BackendOption,
```

Reply:
Makes sense. Updated.